browser/extensions/api/brave_rewards_api.cc

```diff
@@ -661,15 +662,13 @@ ExtensionFunction::ResponseAction BraveRewardsSaveAdsSettingFunction::Run() {
   Profile* profile = Profile::FromBrowserContext(browser_context());
   RewardsService* rewards_service =
       RewardsServiceFactory::GetForProfile(profile);
-  AdsService* ads_service = AdsServiceFactory::GetForProfile(profile);
 
-  if (!rewards_service || !ads_service) {
+  if (!rewards_service) {
     return RespondNow(Error("Service is not initialized"));
   }
 
   if (params->key == "adsEnabled") {
-    const auto is_enabled =
-        params->value == "true" && ads_service->IsSupportedLocale();
+    const auto is_enabled = params->value == "true" && ads::IsSupported();
```

Comment:
so, the problem here is that we still have a rewards -> ads dependency. Ads can depend on rewards, but not vice versa and moving it into bat/ads doesn't really help the situation because eventually that will go away. If there are methods shared by both, they should go in rewards, not ads

Reply:
I did not want the changes to spiral out of control and have raised other issues to continue to resolve these layering violations